Detectors/TPC/simulation/include/TPCSimulation/DigitPad.h

```diff
@@ -85,12 +97,16 @@ DigitPad::DigitPad(int pad)
 inline 
 void DigitPad::setDigit(size_t trackID, float charge)
 {
+  static FairRootManager *mgr = FairRootManager::Instance();
+  const int eventID = mgr->GetEntryNr();
+#ifdef TPC_DIGIT_USEFAIRLINKS
+  mMCLinks.AddLink(FairLink(-1, eventID, "MCTrack", trackID));
+#else
   /// the MC ID is encoded such that we can have 999,999 tracks
   /// numbers larger than 1000000 correspond to the event ID
   /// i.e. 12000010 corresponds to event 12 with track ID 10
   /// \todo Faster would be a bit shift
-#ifdef TPC_DIGIT_USEFAIRLINKS
-  mMCLinks.AddLink(FairLink(-1, FairRootManager::Instance()->GetEntryNr(), "MCTrack", trackID));
+  ++mMCID[(eventID)*1000000 + trackID];
```

Comment:
Why not simply using a struct as the key?

struct MCID{
  unsigned int eventId : 15; // 32k event Id possible 
  unsigned int trackId: 17; // 128K tracks possible
};

Also why using a map rather than a sorted vector with key and value in the same bitfield?

```C++
struct MCIDValue {
  unsigned int eventId : 15; // 32k event Id possible 
  unsigned int trackId: 17; // 128K tracks possible
  unsigned int occurences : 32; // 4GB occurrences possible   
}
std::vector<MCID> mMCID;
```

this has the added value you can use less bits for the occurrences and should be almost two orders of magnitude more compact.